rhombus/private/syntax-class-syntax.rkt

```diff
@@ -1,30 +1,58 @@
 #lang racket/base
 (require (for-syntax racket/base
                      syntax/parse
-                     racket/list
                      racket/set)
          syntax/parse
          (submod "quasiquote.rkt" convert)
          (submod "syntax-class.rkt" for-syntax-class-syntax)
          "definition.rkt"
-         "name-root.rkt")
+         "name-root.rkt"
+         "parse.rkt")
 
-(provide syntax)
+(provide (rename-out [rhombus-syntax syntax]))
 
-(define-simple-name-root syntax
+(define-simple-name-root rhombus-syntax
   class)
 
 (define-for-syntax (generate-pattern-and-attributes stx)
   (syntax-parse stx
-    #:datum-literals (alts group quotes)
+    #:datum-literals (alts group quotes block)
     [(block (group (quotes in-quotes)))
      (define-values (p idrs sidrs can-be-empty?)
        (convert-pattern #:splice? #t
                         #'in-quotes))
      (with-syntax ([((attr ...) ...)
                     (map (lambda (binding) (cons '#:attr binding)) idrs)])
        (values #`(pattern #,p attr ... ...)
-               (map (lambda (binding) (syntax-e (car (syntax->list binding)))) idrs)))]))
+               (map (lambda (binding) (syntax-e (car (syntax->list binding)))) idrs)))]
+    [(block (group (quotes in-quotes)
+                   (block groups ...)))
+     (define-values (p idrs sidrs can-be-empty?)
+       (convert-pattern #:splice? #t
+                        #'in-quotes))
+     (define explicit-attrs
+       (for/fold ([other-forms null]
+                  [attrs null]
+                  #:result (reverse attrs))
+                 ([g (in-list (syntax->list #'(groups ...)))])
+         (syntax-parse g
+           #:datum-literals (group block)
+           [(group #:attr attr-id (block in-block ...))
+            (with-syntax ([(other ...) (reverse other-forms)]
+                          [((id id-ref) ...) (append idrs attrs)])
+              (values
+               other-forms
```

Comment:
I think you meant `null` here, otherwise the forms already added via `other ...` get duplicated

Reply:
I'm not sure if we will want that. If I replace `other-forms` with `null` here then it won't let any explicit attrs refer to earlier definitions. For example 
```
  | '~foo $x':
      def c:
        '14'
      def g:
        24 + unwrap_syntax(c) + unwrap_syntax(x)
      ~attr y:
        '$g'
      def a:
        5 + unwrap_syntax(y) + unwrap_syntax(c)
```
If we use `null`, this won't work because after `~attr y` the definition of `c` is not going to continue to be available for `a`.

Reply:
I see the problem now. Let's discuss a better approach to this.